client/core/core_test.go

```diff
@@ -4287,6 +4299,56 @@ func TestHandleTradeSuspensionMsg(t *testing.T) {
 	}
 }
 
+func setupOrderNotificationVerification(tCore *Core, t *testing.T) func() *OrderNote {
+	ch := tCore.NotificationFeed()
+
+	getNotification := func(tag string) interface{} {
+		t.Helper()
+		select {
+		case n := <-ch:
+			return n
+			// When it works, it should be virtually instant, but I have seen it fail
+			// at 1 millisecond.
+		case <-time.NewTimer(time.Second).C:
+			t.Fatalf("timed out waiting for %s notification", tag)
+		}
+		return nil
+	}
+	getOrderNote := func() *OrderNote {
+		t.Helper()
+		var orderNote *OrderNote
+		for orderNote == nil {
+			ntfn := getNotification("feepayment or balance")
+			switch note := ntfn.(type) {
+			case *OrderNote:
+				orderNote = note
+			}
+		}
+		return orderNote
+	}
+	return getOrderNote
+}
+
+func verifyRevokeNotification(getOrderNote func() *OrderNote, expectedOrderNote *OrderNote, t *testing.T) {
```

Comment:
Notification subjects should be unique, so that's the only thing needed for comparison.